pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -666,8 +666,7 @@ public void deleteCursorComplete(Object ctx) {
                 @Override
                 public void deleteCursorFailed(ManagedLedgerException exception, Object ctx) {
                     log.error("[{}] Failed to delete cursor {}", topic, name);
-                    // Connect the producers back
-                    replicators.get(remoteCluster).startProducer();
+                    replicators.remove(remoteCluster);
```

Comment:
Just one thing here. If the delete cursor fails during the topic load, then the topic load fails and that is fine since it will be re-tried anyway.

If it fails after a policies change though, it will not be retried. 
I think the whole `PersistentTopic.checkReplication()` should be rescheduled after a while to make sure we clean up the cursor properly.

Just retrying the cursor delete could be dangerous, since the remote cluster could be added again before we retry, but the `checkReplication()` is idempotent and uses the latest configuration.